src/utils/validateInput.ts

```diff
@@ -0,0 +1,31 @@
+export type InputValidator<T> = (input: T) => boolean
+
+const passesAll = <T>(iv: InputValidator<T>[]) => (value: T) =>
+  iv.map(fn => fn(value)).reduce((p, c) => p && c, true)
+
+const isStringLengthValid = (maxLen: number): InputValidator<string> => (
```

Comment:
The types for `input` and return can be inferred from `InputValidator<string>` and are redundant.